server/routes/dates.js

```diff
@@ -0,0 +1,111 @@
+const express = require('express')
+const moment = require('moment')
+
+const cardData = require('../db/cardData')
+const graph = require('../db/graph')
+const activities = require('../db/activities')
+
+const router = express.Router()
+
+module.exports = router
+
+// router.get('/cards', (req, res) => {
+//   cardData.getCards()
+// })
+
+router.post('/', (req, res) => {
+  // req.body required to look like :
+  // {userId: 1, records:[{activityId:1, rating:1, log:'blah'},{more cards}]}
+  const {userId, records} = req.body
+
+  // if (isRecordOkay) {
+  cardData.addDate({user_id: userId})
+    .then(dateId => {
+      for (let rec of records) {
+        rec.date_id = dateId[0]
+        rec.activity_id = rec.activityId
+        delete rec.activityId
```

Comment:
You could also use something like:
```js
const parsedRecords = records.map(r => {
  // rec is now everything _except_ activityId
  const { activityId, ...rec } = r

  return {
    date_id: dateId[0],
    activity_id: activityId,
    ...rec
  })
```
There are other approaches to converting camel to snake case, including [lodash](https://lodash.com/docs/4.17.11#snakeCase).